File: tableRazorAssigment/wwwroot/js/user-dto.js
export class UserDto {
    constructor(id, name, userEmail, isUserBlocked, isUserEmailConfirmed, lastSeen, title = null) {
        this.id = id;
        this.name = name;
        this.userEmail = userEmail;
        this.isUserBlocked = isUserBlocked;
        this.isUserEmailConfirmed = isUserEmailConfirmed;
        this.lastSeen = lastSeen instanceof Date ? lastSeen : new Date(lastSeen);
        this.title = title;
    }

    get status() {
        if (this.isUserBlocked) return 'Blocked';
        if (!this.isUserEmailConfirmed) return 'Not verified';
        return 'Active';
    }

}

File: tableRazorAssigment/Model/UserDTO.cs
namespace tableRazorAssigment.Model;

public class UserDto
{
    public string Id { get; set; }

    public string? Title { get; set; }

    public string Name { get; set; }

    public string UserEmail { get; set; }

    public bool IsUserBlocked { get; set; }

    public bool IsUserEmailConfirmed { get; set; }

    public DateTime LastSeen { get; set; }

}